verifiers/internal/gha/provenance_test.go

```diff
@@ -1182,3 +1187,113 @@ func Test_VerifyVersionedTag(t *testing.T) {
 		})
 	}
 }
+
+func Test_VerifyTrustedProvenance(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name                 string
+		envelopePath         string
+		provenanceOpts       *options.ProvenanceOpts
+		trustedBuilderIDName string
+		byob                 bool
+		expectedID           *string
+		expected             error
+	}{
+		{
+			name:         "Verify Trusted (slsa-github-generator) Bazel Builder (v1.8.0",
+			envelopePath: "bazel-trusted-dsseEnvelope.build.slsa",
+			provenanceOpts: &options.ProvenanceOpts{
+				ExpectedBranch:         nil,
+				ExpectedTag:            nil,
+				ExpectedVersionedTag:   nil,
+				ExpectedDigest:         "caaadba2846905ac477c777e96a636e1c2e067fdf6fed90ec9eeca4df18d6ed9",
+				ExpectedSourceURI:      "github.com/enteraga6/slsa-lvl3-generic-provenance-with-bazel-example",
+				ExpectedBuilderID:      "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/delegator_lowperms-generic_slsa3.yml@refs/tags/v1.8.0",
+				ExpectedWorkflowInputs: map[string]string{},
+			},
+			byob:                 true,
+			trustedBuilderIDName: "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/delegator_lowperms-generic_slsa3.yml@refs/tags/v1.8.0",
+			expectedID:           nil,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Re-initializing variable so it is not changed while executing the closure below
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			trustedBuilderID, tErr := utils.TrustedBuilderIDNew(tt.trustedBuilderIDName, true)
+			if tErr != nil {
+				t.Errorf("Provenance Verification FAILED. Error: %v", tErr)
+			}
+
+			envelopeBytes, err := os.ReadFile(filepath.Join("testdata", tt.envelopePath))
+			if err != nil {
+				panic(fmt.Errorf("os.ReadFile: %w", err))
```

Comment:
let's use `t.Errorf` since we're using it in other parts. I know, our existing tests are not consistent about this!

Reply:
Switched =)